app/models/reports/schema_v2.rb

```diff
@@ -191,9 +236,13 @@ def visited_without_bp_taken(with_ltfu: false)
       end
     end
 
-    def percentage(numerator, denominator)
-      return 0 if numerator.nil? || denominator.nil? || denominator == 0 || numerator == 0
-      ((numerator.to_f / denominator) * 100).round(PERCENTAGE_PRECISION)
+    memoize def appts_scheduled_rates(entry)
+      rounded_percentages({
+        appts_scheduled_0_to_14_days_rates: appts_scheduled_0_to_14_days[entry.region.slug][entry.period],
+        appts_scheduled_15_to_30_days_rates: appts_scheduled_15_to_30_days[entry.region.slug][entry.period],
+        appts_scheduled_31_to_60_days_rates: appts_scheduled_31_to_60_days[entry.region.slug][entry.period],
+        appts_scheduled_more_than_60_days_rates: appts_scheduled_more_than_60_days[entry.region.slug][entry.period]
+      })
```

Comment:
I think this sort of think could probably go into a view_component that is dedicated to just the new graph, and that way the  rounded logic could live in a higher layer that the view_component uses.  Also see my other comment on this =).